Honor --quiet parameter properly in fish shell

philipsd6 opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
This is a tiny cosmetic change, as it has been bothering me that my startup configuration output isn't clean when executing `source ~/ansible/hacking/env-setup.fish --quiet` as expected.

* The `^` stderr redirection is deprecated

fish-shell has moved away from using `^` for stderr redirection. See [fish-shell#6192](https://github.com/fish-shell/fish-shell/issues/6192#issuecomment-542363561) and [fish-shell#6206](https://github.com/fish-shell/fish-shell/pull/6206) for details.

* Setting `-q` for `setup.py` in `gen_egg_info` is redundant

When in quiet mode, the complete output of the `gen_egg_info` function is redirected, so adding the `-q` flag when calling `setup.py` in the function is redundant. This is symmetrical with the Bash environment setup script as well.

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
- Installation
  - Running the `devel` branch from a clone (`hacking/env-setup.fish`)

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
**Before:**
```
philipsd6@vm001 ~/ansible (devel)> source hacking/env-setup.fish --quiet
warning: no files found matching 'SYMLINK_CACHE.json'
warning: no previously-included files found matching 'docs/docsite/rst_warnings'
warning: no previously-included files found matching 'docs/docsite/rst/conf.py'
warning: no previously-included files found matching 'docs/docsite/rst/index.rst'
warning: no previously-included files matching '*' found under directory 'docs/docsite/_build'
warning: no previously-included files matching '*.pyc' found under directory 'docs/docsite/_extensions'
warning: no previously-included files matching '*.pyo' found under directory 'docs/docsite/_extensions'
warning: no files found matching '*.ps1' under directory 'lib/ansible/modules/windows'
warning: no files found matching 'validate-modules' under directory 'test/lib/ansible_test/_util/controller/sanity/validate-modules'
find: paths must precede expression: `^'
```
**After:**
```
philipsd6@vm001 ~/ansible (fix/env-setup-fish)> source hacking/env-setup.fish --quiet
philipsd6@vm001 ~/ansible (fix/env-setup-fish)>
```

Commit:
Honor `--quiet` parameter properly in fish shell

* The `^` stderr redirection is deprecated
* Setting `-q` for `setup.py` in `gen_egg_info` is redundant

jborean93 commented:
Just as an FYI it shouldn't be necessary to use the env-setup files anymore. Ansible now supports things like `pip install -e .` to set everything up properly for you. This should be present in the latest devel branch but not older branches. Are you able to try that out as eventually we will be wanting to remove these files.

I unfortunately don't know much about `fish` to validate these changes.

philipsd6 commented:
> Just as an FYI it shouldn't be necessary to use the env-setup files anymore. Ansible now supports things like `pip install -e .` to set everything up properly for you. This should be present in the latest devel branch but not older branches. Are you able to try that out as eventually we will be wanting to remove these files.
> 
> I unfortunately don't know much about `fish` to validate these changes.

Good point, but I've been caught in pip hell doing it that way on more than one occasion as I also switch among python versions. Nevertheless, since I spend most of my time on the devel branch, I'll start switching to that model. Drop this PR or merge as you wish.

jborean93 commented:
If you are able to create a changelog fragment I am happy to merge it in just for the sake of keeping things running as they are while a transition to editable installs happen. Docs for how to do this can be found at https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment.

philipsd6 commented:
> If you are able to create a changelog fragment I am happy to merge it in just for the sake of keeping things running as they are while a transition to editable installs happen. Docs for how to do this can be found at https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment.

Thanks for the reminder! I've added a changelog fragment, and rebased. I used scope=hacking as that's the directory, but maybe it should be env-setup? Let me know.

jborean93 commented:
Thanks for that,
